packages/dds/tree/src/shared-tree-core/editManagerCodecs.ts

```diff
@@ -55,7 +55,7 @@ export function makeEditManagerCodec<TChangeset>(
 	EditManagerEncodingContext
 > {
 	const family = makeEditManagerCodecs(changeCodecs, revisionTagCodec, options);
-	return makeVersionDispatchingCodec(family, { ...options, writeVersion });
+	return makeStrictVersionCodec(family, { ...options, writeVersion });
```

Comment:
Only applied to the EditManagerCodec for PoC purposes, but this same kind of change would need to be applied to all codecs.